### PR TITLE
skip one character on empty match.

### PR DESCRIPTION
**Pull request by L3MON4D3**
Hi, I'm still working on the lsp-parser for luasnip and noticed that `global` is a bit broken right now (empty matches are not accepted at all).
I've modeled my solution to this after quickjs (see [here](https://github.com/bellard/quickjs/blob/2788d71e823b522b178db3b3660ce93689534e6d/quickjs.c#L42857-L42869), where, if I'm understanding everything correctly, the character after the empty match is skipped, to prevent matching it again.)
This works, a downside is that matches sometimes have an empty match appended (eg. `s/(.*)/g` on `a` -> `{[1,2], [2,2]}`), but this seems to be consistent with some quick testing I did via https://rcode5.github.io/jsregex/ and vscode itself

https://user-images.githubusercontent.com/41961280/178162489-fa0b9a31-77d0-4e8c-9a33-2848c9d30a6a.mp4

(you can see the `a` from the second match at the end)

What do you think? It's certainly not pretty, and will have to be adapted a bit for unicode, but it works.

### CODE REVIEW

File: jsregexp.c [hunk 1]
```diff
@@ -32,14 +32,18 @@ static int regex_closure(lua_State *lstate)
...
-		// happens on a regex like (\\s*), don't know how we should handle it
-		// and can't remember how vscode handles it
 		if (capture[0] == capture[1]) {
-			break;
+			// empty match -> continue matching from next character (to prevent an endless loop).
+			// This is basically the same implementation as in quickjs, see
+			// https://github.com/bellard/quickjs/blob/2788d71e823b522b178db3b3660ce93689534e6d/quickjs.c#L42857-L42869
+
+			// +1 works for ascii, take a closer look for unicode
+			++cindex;
```
Comment (kmarius) [inline jsregexp.c:41]: Wouldn't you still want to break/do the next iteration here? Would that help with the empty match being added?

Comment (L3MON4D3) [reply jsregexp.c:41]: Well we do still want a match on a completely empty string, eg. "" should match `(.*)`, so we can't just `break`, the match should be pushed

Comment (kmarius) [reply jsregexp.c:41]: I see. I also checked with vscode to be sure and it does the same.

**Comment by kmarius** [comment]
Thanks, merged.

**Comment by L3MON4D3** [comment]
Nice, thank you👍

**Comment by kmarius** [comment]
> Oh, could you make a new release with this patch?

Done.

**Comment by L3MON4D3** [comment]
Ty! 👍👍

**Pull request merged**